sources/Debate-frontend/src/app/debate/add-update-debate/add-update-debate.component.ts

```diff
@@ -0,0 +1,94 @@
+import { Component, OnInit, Input, ViewChild } from '@angular/core';
+import { AddUpdateDebateDto } from 'src/app/dto/addUpdateDebate.dto';
+import { ENTER, COMMA } from '@angular/cdk/keycodes';
+import { MatChipInputEvent } from '@angular/material/chips';
+import { DebateService } from 'src/app/services/debate.service';
+import { Router } from '@angular/router';
+
+@Component({
+  selector: 'app-add-update-debate',
+  templateUrl: './add-update-debate.component.html',
+  styleUrls: ['./add-update-debate.component.scss']
+})
+export class AddUpdateDebateComponent implements OnInit {
+
+  model: AddUpdateDebateDto;
+  readonly separatorKeysCodes: number[] = [ENTER, COMMA];
+  imageUrl: any;
+  imageData: File;
+  imageInvalid: boolean = false;
+
+  @ViewChild("tagList") tagList;
+  @ViewChild("debateForm") debateForm: HTMLFormElement;
+
+  constructor(private debateService: DebateService, private router: Router) { 
+    this.model = {title: "", description: "", mainTags: [], allTags: ["polityka"]};
+  }
+
+  ngOnInit(): void {
+  }
+
+  onSubmit(){
+    if (!this.debateForm.form.valid || !this.isTagListValid){
+      return;
+    }
+    this.model.mainTags = this.model.allTags.slice(0,3);
+    this.debateService.addDebate(this.model, this.imageData).subscribe(
+      res => this.router.navigate(["/home"]), 
+      err => console.log(err)
+    );
+  }
+
+  get diagnostics(){
+    return JSON.stringify(this.model);
+  }
+
+  addTag(event: MatChipInputEvent){
+      if ((event.value || '').trim()){
+        const value = event.value.trim().toLowerCase();
+        if (this.model.allTags.indexOf(value) >= 0){
+          return;
+        }
+        this.model.allTags.push(value);
+        if (this.model.allTags.length >= 3 && this.model.allTags.length < 7){
+          this.tagList.errorState = false;
+        }else if (this.model.allTags.length > 7){
+          this.tagList.errorState = true;
+        }
+      }
+      if (event.input){
+        event.input.value = "";
+      }    
+  }
+
+  removeTag(tag: string): void {
+    const index = this.model.allTags.indexOf(tag);
+    if (index >= 0) {
+      this.model.allTags.splice(index, 1);
+    }
+
+    if (this.model.allTags.length < 3 || this.model.allTags.length > 7){
+        this.tagList.errorState = true;         
+    } else {
+      this.tagList.errorState = false;
+    }
+  }
+
+  onFileChanged(fileEvent: any){
```

Comment:
```suggestion
  onFileChanged(fileEvent: Event){
```
Add concrete parameter type